### PR TITLE
fix: [CDS-86670]: Handled invalid yaml for ServiceOverrides

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary

**Issue:**: Page breaks with invalid service override YAML.
**Fix:** Parse yam with try catch and show parsing error.

#### Screenshots
**Before:**: YAML parser throws an error for invalid YAML and page breaks.
Screenshot 2023-12-12 at 6 29 54 PM

**After:**
Screenshot 2023-12-12 at 6 06 52 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx [hunk 1]
```diff
@@ -342,259 +351,273 @@ export function ServiceOverrides(): React.ReactElement {
...
+                    <>
+                      <Text intent={Intent.DANGER} font={{ weight: 'bold' }}>
+                        {getString('navigationYamlErrorTitle')}
+                      </Text>
+                      <Text intent={Intent.DANGER}>
+                        {getString('error')}:<pre>{yamlParsingErrorMessage}</pre>
+                      </Text>
+                    </>
+                  ) : (
+                    <Layout.Vertical spacing="medium">
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx:364]: This is just an indentation change.

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx [hunk 2]
```diff
@@ -324,7 +333,7 @@ export function ServiceOverrides(): React.ReactElement {
                 (serviceObject: ServiceResponse) => serviceRef === serviceObject.service?.identifier
               ),
               'service.name',
-              serviceRef
+              serviceOverride.serviceRef
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx:336]: Use fallback value from serviceOverride to handle invalid YAML case

File: src/modules/10-common/strings/strings.en.yaml [hunk 3]
```diff
@@ -1497,7 +1497,7 @@ python: Python
 networkMap: Network Map
 successfullyAdded: Successfully added
 successfullyUpdated: Successfully updated
-errorOccured: 'Error occured while {{verb}} {{entity}}'
+errorOccured: 'Error occurred while {{verb}} {{entity}}'
```
Comment (pankajsingh-harness) [inline src/modules/10-common/strings/strings.en.yaml:1500]: Spell mistake fix

**Comment by pankajsingh-harness** [comment]
trigger messagecheck

**Comment by pankajsingh-harness** [comment]
trigger messagecheck

**Review by rameshx — APPROVED** [review]

**Review by chetanvekariya-harness — COMMENTED** [review]
Can we add jest?

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Pull request merged**